ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java

```diff
@@ -153,59 +142,14 @@ StructuredDataSource getResultDatasource(final Select select, final String name)
   private static PlanNode buildQueryLogicalPlan(
       final String sqlExpression,
       final Query query,
-      final MetaStore tempMetaStore,
+      final MetaStore metaStore,
       final KsqlConfig config
   ) {
-    final QueryAnalyzer queryAnalyzer = new QueryAnalyzer(
-        tempMetaStore,
-        tempMetaStore,
-        config
-    );
+    final QueryAnalyzer queryAnalyzer = new QueryAnalyzer(metaStore, config);
+
     final Analysis analysis = queryAnalyzer.analyze(sqlExpression, query);
     final AggregateAnalysis aggAnalysis = queryAnalyzer.analyzeAggregate(query, analysis);
-    final PlanNode logicalPlan
-        = new LogicalPlanner(analysis, aggAnalysis, tempMetaStore).buildPlan();
-    if (logicalPlan instanceof KsqlStructuredDataOutputNode) {
-      final KsqlStructuredDataOutputNode ksqlStructuredDataOutputNode =
-          (KsqlStructuredDataOutputNode) logicalPlan;
-
-      final StructuredDataSource
-          structuredDataSource =
-          (ksqlStructuredDataOutputNode.getNodeOutputType() == DataSourceType.KTABLE)
-              ? new KsqlTable<>(
-                  sqlExpression,
-                  ksqlStructuredDataOutputNode.getId().toString(),
-                  ksqlStructuredDataOutputNode.getSchema(),
-                  ksqlStructuredDataOutputNode.getKeyField(),
-                  ksqlStructuredDataOutputNode.getTimestampExtractionPolicy(),
-                  ksqlStructuredDataOutputNode.getKsqlTopic(),
-                  "", // Placeholder
-                  Serdes.String()  // Placeholder
-              )
-              : new KsqlStream<>(
-                  sqlExpression,
-                  ksqlStructuredDataOutputNode.getId().toString(),
-                  ksqlStructuredDataOutputNode.getSchema(),
-                  ksqlStructuredDataOutputNode.getKeyField(),
-                  ksqlStructuredDataOutputNode.getTimestampExtractionPolicy(),
-                  ksqlStructuredDataOutputNode.getKsqlTopic(),
-                  Serdes.String()  // Placeholder
-              );
-
-      if (analysis.isDoCreateInto()) {
-        try {
-          tempMetaStore.putTopic(ksqlStructuredDataOutputNode.getKsqlTopic());
-        } catch (final KsqlException e) {
-          final String sourceName = tempMetaStore.getSourceForTopic(
-              ksqlStructuredDataOutputNode.getKsqlTopic().getName()).get().getName();
-          throw new KsqlException(
-              String.format("Cannot create the stream/table. "
-                  + "The output topic %s is already used by %s",
-                  ksqlStructuredDataOutputNode.getKsqlTopic().getKafkaTopicName(), sourceName), e);
-        }
-        tempMetaStore.putSource(structuredDataSource.cloneWithTimeKeyColumns());
-      }
-    }
-    return logicalPlan;
+
+    return new LogicalPlanner(analysis, aggAnalysis, metaStore).buildPlan();
```

Comment:
the code removed is no longer needed as each statement is fully processed before the next is attempted.  This means such placeholders are no longer needed.